pxtblocks/fields/field_sprite.ts

```diff
@@ -63,7 +65,11 @@ namespace pxtblockly {
         }
 
         protected getValueText(): string {
-            if (this.asset && !this.isTemporaryAsset()) return `assets.image\`${this.asset.meta.displayName || this.asset.id.substr(this.asset.id.lastIndexOf(".") + 1)}\``;
+            if (this.asset && !this.isTemporaryAsset()) {
+                const id = this.asset.meta.displayName || this.asset.id.substr(this.asset.id.lastIndexOf(".") + 1);
```

Comment:
hmm should the no displayName fallback for tiles match the one in field_tileset? i just put out this pr: https://github.com/microsoft/pxt/pull/7855/files but maybe we should refactor this to be someplace central?